iocore/net/P_NetAccept.h

```diff
@@ -73,7 +73,11 @@ struct NetAcceptAction : public Action, public RefCountObj {
     return Action::operator=(acont);
   }
 
-  ~NetAcceptAction() override { Debug("net_accept", "NetAcceptAction dying"); }
+  ~NetAcceptAction() override
+  {
+    static DbgCtl dbg_ctl{"net_accept"};
```

Comment:
Are there any concerns about late initialization for this? IIRC the compiler can arrange for `dbg_ctl` to be constructed on first invocation rather than process start.

Reply:
I used the reference count work-around for the C++ initialization/destruction order fiasco.  Construction of a DbgCtl increments the reference count to the tag container.  The tag container is created when the reference count goes from 0 to 1:

https://github.com/apache/trafficserver/blob/2acf9a47647c1e0374257643a21854dcefa1f2ea/include/tscore/DbgCtl.h#L38
https://github.com/apache/trafficserver/blob/2acf9a47647c1e0374257643a21854dcefa1f2ea/src/tscore/DbgCtl.cc#L123
https://github.com/apache/trafficserver/blob/2acf9a47647c1e0374257643a21854dcefa1f2ea/src/tscore/DbgCtl.cc#L72

I probably would have gotten a promotion for figuring all of this out, if I were under 40 and it involved YAML.

Reply:
Also note that the static local DbgCtl is key to getting the old Debug() call to have the improved performance:
https://github.com/apache/trafficserver/blob/9cd7e5798dca2ed299ecb998b369bda0ab084295/include/tscore/Diags.h#L218
That code has gone through both years of review and running in production.

Reply:
If you want to see something really scary, pull over, here's a static DbgCtl that's local to a lambda:
https://github.com/apache/trafficserver/blob/9cd7e5798dca2ed299ecb998b369bda0ab084295/include/tscore/Diags.h#L240

Reply:
Yes because the constructor of _RegistryAccessor takes the registry's mutex:
https://github.com/apache/trafficserver/blob/fc76b290472e671d628396fe0367b3b0c8c2d4c8/src/tscore/DbgCtl.cc#L139